kubernetes/kubernetes_service.go

```diff
@@ -76,41 +85,24 @@ func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (
 	}
 	serviceDetails.Endpoints = endpointsResponse.endpoints
 
-	podNames := make([]string, 0)
-	for _, subset := range serviceDetails.Endpoints.Subsets {
-		for _, address := range subset.Addresses {
-			targetRef := address.TargetRef
-			if targetRef != nil && targetRef.Kind == "Pod" {
-				podNames = append(podNames, targetRef.Name)
-			}
-		}
+	deploymentsResponse := <-deploymentsChan
+	if deploymentsResponse.err != nil {
+		return nil, deploymentsResponse.err
 	}
+	serviceDetails.Deployments = deploymentsResponse.deployments
 
-	podChan := make(chan podResponse, len(podNames))
-	var wg sync.WaitGroup
-
-	for _, podName := range podNames {
-		wg.Add(1)
-		go func(podName string) {
-			defer wg.Done()
-			log.Infof("podName %v", podName)
-			pod, err := in.k8s.CoreV1().Pods(namespace).Get(podName, emptyGetOptions)
-			podChan <- podResponse{pod: pod, err: err}
-		}(podName)
-	}
+	return &serviceDetails, nil
+}
 
-	go func() {
-		wg.Wait()
-		close(podChan)
-	}()
+func filterByService(serviceName string, dl *v1beta1.DeploymentList) *v1beta1.DeploymentList {
+	var deployments []v1beta1.Deployment
 
-	serviceDetails.Pods = make([]*v1.Pod, 0)
-	for podResponse := range podChan {
-		if podResponse.err != nil {
-			return nil, podResponse.err
+	for _, deployment := range dl.Items {
+		if deployment.ObjectMeta.Labels != nil && deployment.ObjectMeta.Labels["app"] == serviceName {
```

Comment:
This is good. Nothing to touch now.
But in the future this can be configurable, as label used to define an app is open.
i.e. it can be "App", "Application", or whatever.
So, it something to don't forget.
In the previous Pods functionality, relationship comes from endpoints, which is the only "hard" link.
Just leaving a comment here to not forget and sync in the same topic.

Reply:
:100: